backend/packages/Upgrade/src/api/services/ScheduledJobService.ts

```diff
@@ -6,57 +6,106 @@ import { ScheduledJob, SCHEDULE_TYPE } from '../models/ScheduledJob';
-        return experimentService.updateState(scheduledJob.experiment.id, EXPERIMENT_STATE.ENROLLING, systemUser);
+    return await getConnection().transaction(async (transactionalEntityManager) => {
+      try {
+        const scheduledJobRepository = transactionalEntityManager.getRepository(ScheduledJob);
+        const scheduledJob = await scheduledJobRepository.findOne(id, { relations: ['experiment'] });
+
+        const currentDate = new Date();
+        const timeDif = Math.abs(currentDate.getTime() - scheduledJob.timeStamp.getTime());
+        const fiveHoursInMS = 18000000;
+
+        if (timeDif > fiveHoursInMS) {
+          const errorMsg =  'Time Differnce of more than 5 hours is found';
+          await scheduledJobRepository.delete({id: scheduledJob.id});
+          throw new Error(errorMsg);
+        }
+
+        if (scheduledJob && scheduledJob.experiment) {
+          const experimentRepository = transactionalEntityManager.getRepository(Experiment);
+          const experiment = await experimentRepository.findOne(scheduledJob.experiment.id);
+          if (scheduledJob && experiment) {
+            const systemUser = await transactionalEntityManager
+              .getRepository(User)
+              .findOne({ email: systemUserDoc.email });
+            const experimentService = Container.get<ExperimentService>(ExperimentService);
+            // update experiment startOn
+            await experimentRepository.update({ id: experiment.id }, { startOn: null });
+            return experimentService.updateState(
+              scheduledJob.experiment.id,
+              EXPERIMENT_STATE.ENROLLING,
+              systemUser,
+              null,
+              transactionalEntityManager
+            );
+          }
+        }
+        return {};
+      } catch (error) {
+        this.log.error('Error in start experiment of schedular ', error.message);
+        return error;
       }
-    }
-    return {};
+    });
   }
 
   public async endExperiment(id: string): Promise<any> {
-    const scheduledJob = await this.scheduledJobRepository.findOne(id, { relations: ['experiment'] });
-    const experiment = await this.experimentRepository.findOne(scheduledJob.experiment.id);
-    if (scheduledJob && experiment) {
-      // get system user
-      const systemUser = await this.userRepository.findOne({ email: systemUserDoc.email });
-      const experimentService = Container.get<ExperimentService>(ExperimentService);
-      // update experiment endOn
-      await this.experimentRepository.update({ id: experiment.id }, { endOn: null });
-      return experimentService.updateState(
-        scheduledJob.experiment.id,
-        EXPERIMENT_STATE.ENROLLMENT_COMPLETE,
-        systemUser
-      );
-    }
-    return {};
+    return await getConnection().transaction(async (transactionalEntityManager) => {
+      try {
+        const schedulerJobRepository = transactionalEntityManager.getRepository(ScheduledJob);
```

Comment:
@RidhamShah Can you please update the name to scheduledJobRepository instead of schedulerJobRepository?

Reply:
@RidhamShah if done then let us know.

Reply:
Code Updated